pkg/virtctl/imageupload/imageupload.go

```diff
@@ -926,3 +936,34 @@ func handleEventErrors(client kubecli.KubevirtClient, pvcName, dvName, namespace
 
 	return nil
 }
+
+func createDataSource(client kubecli.KubevirtClient, name, namespace string) error {
+	if defaultPreference == "" {
+		return fmt.Errorf("Failed to create DataSource %s/%s. Set default preference.", namespace, name)
+	}
```

Comment:
Shouldn't it be possible to create a DS even if no instance type / pref was specified?

Reply:
Yes, but then I think it won't display it in the Web UI because the label is missing? What is the benefit of an existing DS without relevant label?

Reply:
Strictly speaking, there is no UI in KubeVirt. IMO it should be possible to create a DataSource without the labels too.